apps/challenges/views.py

```diff
@@ -1849,6 +1849,62 @@ def create_challenge_using_zip_file(request, challenge_host_team_pk):
         )
     },
 )
+@api_view(["GET"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def challenge_has_finished_submissions(request, challenge_pk):
+    """
+    Checks if all challenge phases have finished submissions for the given challenge
+    """
+    challenge = get_challenge_model(challenge_pk)
+    challenge_phases = ChallengePhase.objects.filter(challenge=challenge)
+    unfinished_phases = []
+
+    for challenge_phase in challenge_phases:
+        submissions = Submission.objects.filter(
+            challenge_phase=challenge_phase,
+            status="finished"
+        )
+
+        if not submissions.exists():
+            unfinished_phases.append(challenge_phase.name)
+
+    if unfinished_phases:
+        error_message = f"The following challenge phases do not have finished submissions: {', '.join(unfinished_phases)}"
+        return Response({"error": error_message}, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    send_slack_approval_request_response = send_slack_approval_request(challenge_pk)
+    if send_slack_approval_request_response["status"] == "success":
+        response_data = {
+            "message": "All challenge phases have finished submissions.",
+        }
+        return Response(response_data, status=status.HTTP_200_OK)
+    else:
+        error_message = f"All challenge phases have finished submissions, but there was an error sending the Slack notification: {send_slack_approval_request_response['message']}"
```

Comment:
"Sorry, there was an error sending approval request. Please try again."

We don't want them to know how we receive requests.